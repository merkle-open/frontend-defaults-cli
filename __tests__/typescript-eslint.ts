/// <reference types="@types/jest" />
import { apiIt } from './shared';

describe('typescript-eslint', () => {
	it('default', async () =>
		await apiIt('typescript-eslint', {
			ts: true,
			typescriptEslint: true,
		}));
	it('with-prettier', async () =>
		await apiIt('typescript-eslint-with-prettier', {
			ts: true,
			typescriptEslint: true,
			prettier: true,
		}));
	it('with-githooks', async () =>
		await apiIt('typescript-eslint-with-githooks', {
			ts: true,
			typescriptEslint: true,
			githooks: true,
		}));
});
