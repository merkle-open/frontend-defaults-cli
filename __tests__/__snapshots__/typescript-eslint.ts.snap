// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript-eslint default 1`] = `
"/.idea/
/node_modules/
/*.js"
`;

exports[`typescript-eslint default 2`] = `
"module.exports = {
	extends: ['@namics/eslint-config/configurations/typescript-react.js'].map(require.resolve),
};
"
`;

exports[`typescript-eslint default 3`] = `
"{
  \\"options\\": {
    \\"ts\\": true,
    \\"es\\": false,
    \\"eslint\\": true,
    \\"editorconfig\\": false,
    \\"prettier\\": false,
    \\"stylelint\\": false,
    \\"gitignore\\": false,
    \\"npmrc\\": false,
    \\"readme\\": false,
    \\"githooks\\": false,
    \\"commitlint\\": false,
    \\"nodeVersion\\": false,
    \\"webpack\\": false,
    \\"build\\": false,
    \\"install\\": false,
    \\"force\\": true,
    \\"dryRun\\": false,
    \\"mode\\": \\"api\\"
  },
  \\"mergedFiles\\": {
    \\"package.json\\": {
      \\"override\\": false,
      \\"data\\": {
        \\"scripts\\": {
          \\"lint\\": \\"npm-run-all lint:*\\",
          \\"lint:ts\\": \\"eslint . --ext .jsx,.js,.ts,.tsx\\"
        },
        \\"devDependencies\\": {
          \\"@namics/eslint-config\\": \\"7.0.2\\",
          \\"@namics/ts-config\\": \\"0.3.0\\",
          \\"eslint\\": \\"6.0.1\\",
          \\"eslint-plugin-import\\": \\"2.18.0\\",
          \\"npm-run-all\\": \\"4.1.5\\",
          \\"typescript\\": \\"3.4.5\\"
        }
      }
    },
    \\".eslintrc.js\\": {
      \\"override\\": false,
      \\"data\\": \\"module.exports = {\\\\n\\\\textends: ['@namics/eslint-config/configurations/typescript-react.js'].map(require.resolve),\\\\n};\\\\n\\"
    },
    \\".eslintignore\\": {
      \\"override\\": false,
      \\"data\\": \\"/.idea/\\\\n/node_modules/\\\\n/*.js\\"
    },
    \\"tsconfig.json\\": {
      \\"override\\": false,
      \\"data\\": \\"{\\\\n\\\\t\\\\\\"extends\\\\\\": \\\\\\"@namics/ts-config\\\\\\",\\\\n\\\\t\\\\\\"exclude\\\\\\": [\\\\\\"dist\\\\\\", \\\\\\"build\\\\\\", \\\\\\"node_modules\\\\\\", \\\\\\"__tests__\\\\\\"]\\\\n}\\\\n\\"
    }
  }
}"
`;

exports[`typescript-eslint default 4`] = `
"{
  \\"scripts\\": {
    \\"lint\\": \\"npm-run-all lint:*\\",
    \\"lint:ts\\": \\"eslint . --ext .jsx,.js,.ts,.tsx\\"
  },
  \\"devDependencies\\": {
    \\"@namics/eslint-config\\": \\"7.0.2\\",
    \\"@namics/ts-config\\": \\"0.3.0\\",
    \\"eslint\\": \\"6.0.1\\",
    \\"eslint-plugin-import\\": \\"2.18.0\\",
    \\"npm-run-all\\": \\"4.1.5\\",
    \\"typescript\\": \\"3.4.5\\"
  }
}"
`;

exports[`typescript-eslint default 5`] = `
"{
	\\"extends\\": \\"@namics/ts-config\\",
	\\"exclude\\": [\\"dist\\", \\"build\\", \\"node_modules\\", \\"__tests__\\"]
}
"
`;

exports[`typescript-eslint with-githooks 1`] = `
"/.idea/
/node_modules/
/*.js"
`;

exports[`typescript-eslint with-githooks 2`] = `
"module.exports = {
	extends: ['@namics/eslint-config/configurations/typescript-react.js'].map(require.resolve),
};
"
`;

exports[`typescript-eslint with-githooks 3`] = `
"{
  \\"options\\": {
    \\"ts\\": true,
    \\"es\\": false,
    \\"eslint\\": true,
    \\"editorconfig\\": false,
    \\"prettier\\": false,
    \\"stylelint\\": false,
    \\"gitignore\\": false,
    \\"npmrc\\": false,
    \\"readme\\": false,
    \\"githooks\\": true,
    \\"commitlint\\": false,
    \\"nodeVersion\\": false,
    \\"webpack\\": false,
    \\"build\\": false,
    \\"install\\": false,
    \\"force\\": true,
    \\"dryRun\\": false,
    \\"mode\\": \\"api\\"
  },
  \\"mergedFiles\\": {
    \\"package.json\\": {
      \\"override\\": false,
      \\"data\\": {
        \\"scripts\\": {
          \\"lint\\": \\"npm-run-all lint:*\\",
          \\"lint:ts\\": \\"eslint . --ext .jsx,.js,.ts,.tsx\\"
        },
        \\"devDependencies\\": {
          \\"@namics/eslint-config\\": \\"7.0.2\\",
          \\"@namics/ts-config\\": \\"0.3.0\\",
          \\"eslint\\": \\"6.0.1\\",
          \\"eslint-plugin-import\\": \\"2.18.0\\",
          \\"husky\\": \\"2.3.0\\",
          \\"lint-staged\\": \\"8.1.7\\",
          \\"npm-run-all\\": \\"4.1.5\\",
          \\"typescript\\": \\"3.4.5\\"
        },
        \\"husky\\": {
          \\"hooks\\": {
            \\"pre-commit\\": \\"lint-staged\\"
          }
        },
        \\"lint-staged\\": {
          \\"linters\\": {
            \\"src/**/*.{js,jsx,ts,tsx}\\": [
              \\"eslint --ext .jsx,.js,.ts,.tsx\\"
            ]
          }
        }
      }
    },
    \\".huskyrc\\": {
      \\"override\\": false,
      \\"data\\": \\"# ~/.huskyrc\\\\n# Note: it will run in /bin/sh shell\\\\nif command -v nodenv >/dev/null 2>&1; then\\\\n  eval \\\\\\"$(nodenv init -)\\\\\\"\\\\nfi\\"
    },
    \\".eslintrc.js\\": {
      \\"override\\": false,
      \\"data\\": \\"module.exports = {\\\\n\\\\textends: ['@namics/eslint-config/configurations/typescript-react.js'].map(require.resolve),\\\\n};\\\\n\\"
    },
    \\".eslintignore\\": {
      \\"override\\": false,
      \\"data\\": \\"/.idea/\\\\n/node_modules/\\\\n/*.js\\"
    },
    \\"tsconfig.json\\": {
      \\"override\\": false,
      \\"data\\": \\"{\\\\n\\\\t\\\\\\"extends\\\\\\": \\\\\\"@namics/ts-config\\\\\\",\\\\n\\\\t\\\\\\"exclude\\\\\\": [\\\\\\"dist\\\\\\", \\\\\\"build\\\\\\", \\\\\\"node_modules\\\\\\", \\\\\\"__tests__\\\\\\"]\\\\n}\\\\n\\"
    }
  }
}"
`;

exports[`typescript-eslint with-githooks 4`] = `
"# ~/.huskyrc
# Note: it will run in /bin/sh shell
if command -v nodenv >/dev/null 2>&1; then
  eval \\"$(nodenv init -)\\"
fi"
`;

exports[`typescript-eslint with-githooks 5`] = `
"{
  \\"scripts\\": {
    \\"lint\\": \\"npm-run-all lint:*\\",
    \\"lint:ts\\": \\"eslint . --ext .jsx,.js,.ts,.tsx\\"
  },
  \\"devDependencies\\": {
    \\"@namics/eslint-config\\": \\"7.0.2\\",
    \\"@namics/ts-config\\": \\"0.3.0\\",
    \\"eslint\\": \\"6.0.1\\",
    \\"eslint-plugin-import\\": \\"2.18.0\\",
    \\"husky\\": \\"2.3.0\\",
    \\"lint-staged\\": \\"8.1.7\\",
    \\"npm-run-all\\": \\"4.1.5\\",
    \\"typescript\\": \\"3.4.5\\"
  },
  \\"husky\\": {
    \\"hooks\\": {
      \\"pre-commit\\": \\"lint-staged\\"
    }
  },
  \\"lint-staged\\": {
    \\"linters\\": {
      \\"src/**/*.{js,jsx,ts,tsx}\\": [
        \\"eslint --ext .jsx,.js,.ts,.tsx\\"
      ]
    }
  }
}"
`;

exports[`typescript-eslint with-githooks 6`] = `
"{
	\\"extends\\": \\"@namics/ts-config\\",
	\\"exclude\\": [\\"dist\\", \\"build\\", \\"node_modules\\", \\"__tests__\\"]
}
"
`;

exports[`typescript-eslint with-prettier 1`] = `
"/.idea/
/node_modules/
/*.js"
`;

exports[`typescript-eslint with-prettier 2`] = `
"module.exports = {
	extends: [
		'@namics/eslint-config/configurations/typescript-react.js',
		'@namics/eslint-config/configurations/typescript-react-disable-styles.js',
	].map(require.resolve),
};
"
`;

exports[`typescript-eslint with-prettier 3`] = `
"{
  \\"options\\": {
    \\"ts\\": true,
    \\"es\\": false,
    \\"eslint\\": true,
    \\"editorconfig\\": false,
    \\"prettier\\": true,
    \\"stylelint\\": false,
    \\"gitignore\\": false,
    \\"npmrc\\": false,
    \\"readme\\": false,
    \\"githooks\\": false,
    \\"commitlint\\": false,
    \\"nodeVersion\\": false,
    \\"webpack\\": false,
    \\"build\\": false,
    \\"install\\": false,
    \\"force\\": true,
    \\"dryRun\\": false,
    \\"mode\\": \\"api\\"
  },
  \\"mergedFiles\\": {
    \\"package.json\\": {
      \\"override\\": false,
      \\"data\\": {
        \\"scripts\\": {
          \\"lint\\": \\"npm-run-all lint:*\\",
          \\"lint:ts\\": \\"eslint . --ext .jsx,.js,.ts,.tsx\\",
          \\"prettier\\": \\"prettier --write \\\\\\"**/*.*(js|jsx|ts|tsx|json)\\\\\\"\\"
        },
        \\"devDependencies\\": {
          \\"@namics/eslint-config\\": \\"7.0.2\\",
          \\"@namics/prettier-config\\": \\"0.2.0\\",
          \\"@namics/ts-config\\": \\"0.3.0\\",
          \\"eslint\\": \\"6.0.1\\",
          \\"eslint-plugin-import\\": \\"2.18.0\\",
          \\"npm-run-all\\": \\"4.1.5\\",
          \\"prettier\\": \\"1.17.1\\",
          \\"typescript\\": \\"3.4.5\\"
        }
      }
    },
    \\".prettierignore\\": {
      \\"override\\": false,
      \\"data\\": \\"# Node\\\\n**/node_modules\\\\n**/package.json\\\\n**/package-lock.json\\\\n\\\\n# Generated\\\\n**/bower.json\\\\n**/lerna.json\\\\n\\\\n# Build\\\\n**/build\\\\n**/dist\\\\n**/public\\\\n**/coverage\\\\n**/storybook-static\\\\n\\\\n# Nitro\\\\n**/project/blueprints\\\\n**/.yo-rc.json\\"
    },
    \\".prettierrc.js\\": {
      \\"override\\": false,
      \\"data\\": \\"module.exports = require('@namics/prettier-config');\\\\n\\"
    },
    \\".eslintrc.js\\": {
      \\"override\\": false,
      \\"data\\": \\"module.exports = {\\\\n\\\\textends: [\\\\n\\\\t\\\\t'@namics/eslint-config/configurations/typescript-react.js',\\\\n\\\\t\\\\t'@namics/eslint-config/configurations/typescript-react-disable-styles.js',\\\\n\\\\t].map(require.resolve),\\\\n};\\\\n\\"
    },
    \\".eslintignore\\": {
      \\"override\\": false,
      \\"data\\": \\"/.idea/\\\\n/node_modules/\\\\n/*.js\\"
    },
    \\"tsconfig.json\\": {
      \\"override\\": false,
      \\"data\\": \\"{\\\\n\\\\t\\\\\\"extends\\\\\\": \\\\\\"@namics/ts-config\\\\\\",\\\\n\\\\t\\\\\\"exclude\\\\\\": [\\\\\\"dist\\\\\\", \\\\\\"build\\\\\\", \\\\\\"node_modules\\\\\\", \\\\\\"__tests__\\\\\\"]\\\\n}\\\\n\\"
    }
  }
}"
`;

exports[`typescript-eslint with-prettier 4`] = `
"# Node
**/node_modules
**/package.json
**/package-lock.json

# Generated
**/bower.json
**/lerna.json

# Build
**/build
**/dist
**/public
**/coverage
**/storybook-static

# Nitro
**/project/blueprints
**/.yo-rc.json"
`;

exports[`typescript-eslint with-prettier 5`] = `
"module.exports = require('@namics/prettier-config');
"
`;

exports[`typescript-eslint with-prettier 6`] = `
"{
  \\"scripts\\": {
    \\"lint\\": \\"npm-run-all lint:*\\",
    \\"lint:ts\\": \\"eslint . --ext .jsx,.js,.ts,.tsx\\",
    \\"prettier\\": \\"prettier --write \\\\\\"**/*.*(js|jsx|ts|tsx|json)\\\\\\"\\"
  },
  \\"devDependencies\\": {
    \\"@namics/eslint-config\\": \\"7.0.2\\",
    \\"@namics/prettier-config\\": \\"0.2.0\\",
    \\"@namics/ts-config\\": \\"0.3.0\\",
    \\"eslint\\": \\"6.0.1\\",
    \\"eslint-plugin-import\\": \\"2.18.0\\",
    \\"npm-run-all\\": \\"4.1.5\\",
    \\"prettier\\": \\"1.17.1\\",
    \\"typescript\\": \\"3.4.5\\"
  }
}"
`;

exports[`typescript-eslint with-prettier 7`] = `
"{
	\\"extends\\": \\"@namics/ts-config\\",
	\\"exclude\\": [\\"dist\\", \\"build\\", \\"node_modules\\", \\"__tests__\\"]
}
"
`;
